src/client/datascience/baseJupyterSession.ts

```diff
@@ -217,7 +219,8 @@ export abstract class BaseJupyterSession implements IJupyterSession {
         if (!this.session?.kernel) {
             throw new SessionDisposedError();
         }
-        return this.session.kernel.requestExecute(content, disposeOnDone, metadata);
+        // Make sure to chain the executes
+        return this.chainingExecute.requestExecute(this.session.kernel, content, disposeOnDone, metadata);
```

Comment:
Question (not sure myself). If this QT event loop can't handle the queuing would we also need this chain on the other request functions here? Like requestDebug and requestInspect?

Reply:
Not sure. It might, but we don't send requestDebug/requestInspect in multiple spots. And requestInspect at the same time as requestExecute does work with QT5. At least I couldn't break it variables open.